File: apps/flutter_app/lib/models/cursor_paged_result.dart
class CursorPagedResult<T> {
  final List<T> items;
  final String? nextCursor;
  final bool hasMore;

  CursorPagedResult({
    required this.items,
    this.nextCursor,
    required this.hasMore,
  });

  factory CursorPagedResult.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic) fromJsonT,
  ) {
    return CursorPagedResult<T>(
      items: (json['items'] as List).map(fromJsonT).toList(),
      nextCursor: json['nextCursor'] as String?,
      hasMore: json['hasMore'] as bool,
    );
  }
}

File: backend/Valora.Application/Common/Models/CursorPagedResult.cs
namespace Valora.Application.Common.Models;

public class CursorPagedResult<T>
{
    public List<T> Items { get; set; } = new();
    public string? NextCursor { get; set; }
    public bool HasMore { get; set; }

    public CursorPagedResult(List<T> items, string? nextCursor, bool hasMore)
    {
        Items = items;
        NextCursor = nextCursor;
        HasMore = hasMore;
    }
}
